src/networking/events.rs

```diff
@@ -0,0 +1,39 @@
+use crate::networking::{id::ConnectionId, id::StreamId, packets::Packet};
+use tokio::sync::mpsc::UnboundedSender;
+
+use super::streams::RecvError;
+
+/// An event generated by the network
+#[derive(Debug)]
+pub enum ReceiveEvent {
```

Comment:
Might be good to document each of these variants, it's kinda confusing to me from here what they do even.